inlet/metadata/provider/root.go

```diff
@@ -9,13 +9,17 @@ import (
 	"net/netip"
 
 	"akvorado/common/reporter"
+	"akvorado/common/schema"
 )
 
 // Interface contains the information about an interface.
 type Interface struct {
-	Name        string `validate:"required"`
-	Description string `validate:"required"`
-	Speed       uint   `validate:"required"`
+	Name         string `validate:"required"`
+	Description  string `validate:"required"`
+	Speed        uint   `validate:"required"`
+	Provider     string
+	Connectivity string
+	Boundary     string
```

Comment:
string? This could be a `schema.InterfaceBoundary`. I suppose this is because it needs to be used in configuration, but that's something you can handle by registering a deserializer.

Reply:
Actually I wasn't sure exactly how to do that properly, especially since it's required for all kinds of sources (ie. config (mapstructure decode ?), http sources (custom code after JQ call ?) and when config is given to inlet from orchestrator (yaml ?).

Also, I didn't like that the `iota` value would be sent over the network from orchestrator to inlet (maybe there is a version missmatch, and the two values don't align) ?

How do you think this could be done properly ?

Reply:
For me, everything should be handled by something like this: https://github.com/akvorado/akvorado/blob/main/common/kafka/config.go#L80-L124.

Reply:
done as well, using your example as a baseline (I never used UnmarshalText before :))